### DISCUSSION TITLE
Dart code metrics

### DISCUSSION THREAD

**Discussion by thomtomdup** [codemagic-ci-cd/codemagic-docs#2060] [Q&A]
### Use case

```Text
Integration of Dart Code Metrics like previous month ago.
```

### Idea

```Text
Could it be possible to add Dart Code Metrics like before for people that have buy a DCM licence like me ?😊

Thank you.
```

**Comment by Ycoi** [comment]
Thanks for the request @thomtomdup! Could you clarify what feature or functionality you exactlyt are missing? As far as I know you should be able to run DCM as before.

**Reply by thomtomdup** [reply]
This was in the test part.
I could select Dart Code Metrics before.
Now i have only flutter "Enable Flutter analyzer"

<img width="1190" alt="Capture d’écran 2023-09-20 à 11 57 52" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/57366945/a2d7c688-37c5-4360-9e12-182c2cf2b4ef">

**Reply by Ycoi** [reply]
Thanks for the clarification! Unfortunately DCM depracated the plugin we were using for the integration as stated here https://github.com/orgs/codemagic-ci-cd/discussions/1767

You should still be able to use DCM by adding a custom Pre-test script which installs and activates DCM to your Codemagic workflow. See the DCM docs for more details https://dcm.dev/docs/getting-started/#running-executable-on-cicd

**Reply by thomtomdup** [reply]
Yes i know that it has been removed from Flutter Workflow Editor but my request is to re-integrate it for people that have licence and to be added in test process. 🙂
If i use script and test in script is not validated , the process run anyway and not stop with error.

**Reply by mikhail-tokarev** [reply]
hey @thomtomdup 
even it's not available in UI you still can use DCM from custom scripts similar to any other tool

> If i use script and test in script is not validated , the process run anyway and not stop with error.

I believe it's an easy fix and if you share build id we will be able to help you with it

**Reply by himesh-cm** [reply]
Hi @thomtomdup, 
Are you still in need of assistance? Please provide your build ID so I can investigate.

**Reply by thomtomdup** [reply]
Hello.
Excuse me i forgot to reply.
My build id is : 651e97afe43a5111382160c4

The process is in the Post-test script result.
It have style issues but the build continue.

Thank you for your support.

**Reply by thomtomdup** [reply]
Hello.
I have replied ( previous reply ). Why is it closed @dtrdic ?
If you need more informations or help I'm available.
Thank you.

**Reply by himesh-cm** [reply]
Hi @thomtomdup, Apologies, the discussion got missed as it was not tagged. I have tagged it and will get back to you by tomorrow.

**Reply by himesh-cm** [reply]
Hey, sorry for the delay, I wanted to inform you all that we are creating a dedicated page in the docs. We will keep you updated once it's live. Thank you for your patience regarding this matter.

**Reply by himesh-cm** [reply]
Hi @thomtomdup  - We have added a guide to our integration section in the documentation. Please check it out [here](https://docs.codemagic.io/integrations/dcm-integration/).

**Reply by thomtomdup** [reply]
Hello.
Thank you.
I have already done this but the request is more to raise issue in test process to stop the build process if the Dart code Metrics send issues.
Is it possible to have it ? It was like this before dart code metrics was removing  from the process.
Thank you.
Regards

**Reply by thomtomdup** [reply]
Hello.
Any news about my request @himesh-cm ? 😊
Thank you.

**Comment by icarusdust** [comment]
I will snooze this thread, and @himesh-cm will re-open it with updates.
